src/yuzu/main.cpp

```diff
@@ -2840,170 +2840,335 @@ void GMainWindow::OnGameListNavigateToGamedbEntry(u64 program_id,
 
-    switch (target) {
-    case GameListShortcutTarget::Desktop: {
-        const QString desktop_path =
-            QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
-        target_directory = desktop_path.toUtf8().toStdString();
-        break;
+#if defined(__linux__) || defined(__FreeBSD__) // Linux and FreeBSD
+    // Reference for the desktop file template:
+    // https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.0.html
+    try {
+
+        // Plus 'Type' is required
+        if (name.empty()) {
+            LOG_ERROR(Frontend, "Name is empty");
+            return false;
+        }
+
+        // Append .desktop extension
+        shortcut_path_full += ".desktop";
+
+        // Create shortcut file
+        std::ofstream shortcut_stream(shortcut_path_full, std::ios::binary | std::ios::trunc);
+
+        if (shortcut_stream.is_open()) {
+
+            fmt::print(shortcut_stream, "[Desktop Entry]\n");
+            fmt::print(shortcut_stream, "Type=Application\n");
+            fmt::print(shortcut_stream, "Version=1.0\n");
+            fmt::print(shortcut_stream, "Name={}\n", name);
+
+            if (!comment.empty()) {
+                fmt::print(shortcut_stream, "Comment={}\n", comment);
+            }
+
+            if (std::filesystem::is_regular_file(icon_path)) {
+                fmt::print(shortcut_stream, "Icon={}\n", icon_path.string());
+            }
+
+            fmt::print(shortcut_stream, "TryExec={}\n", command.string());
+            fmt::print(shortcut_stream, "Exec={}", command.string());
+
+            if (!arguments.empty()) {
+                fmt::print(shortcut_stream, " {}", arguments);
+            }
+
+            fmt::print(shortcut_stream, "\n");
+
+            if (!categories.empty()) {
+                fmt::print(shortcut_stream, "Categories={}\n", categories);
+            }
+
+            if (!keywords.empty()) {
+                fmt::print(shortcut_stream, "Keywords={}\n", keywords);
+            }
+
+            // Flush and close file
+            shortcut_stream.flush();
+            shortcut_stream.close();
+            return true;
+        } else {
+            LOG_ERROR(Frontend, "Failed to create shortcut");
+        }
+        shortcut_stream.close();
+    } catch (const std::exception& e) {
+        LOG_ERROR(Frontend, "Failed to create shortcut: {}", e.what());
     }
-    case GameListShortcutTarget::Applications: {
-        const QString applications_path =
-            QStandardPaths::writableLocation(QStandardPaths::ApplicationsLocation);
-        if (applications_path.isEmpty()) {
-            const char* home = std::getenv("HOME");
-            if (home != nullptr) {
-                target_directory = std::filesystem::path(home) / ".local/share/applications";
+    return false;
+#elif defined(_WIN32) // Windows
+    HRESULT hr = CoInitialize(NULL);
+    if (SUCCEEDED(hr)) {
+
```

Comment:
```suggestion
    if (!SUCCEEDED(hr)) {
        LOG_ERROR(Frontend, "CoInitialize failed");
        return false;
    }

```

And then unindent the rest of the block